### PR TITLE
Allow author data as an object

### PR DESCRIPTION
**Pull request by pathawks**
Try to mirror behavior of **jekyll-feed**

#18 #23

### CODE REVIEW

File: lib/template.html [hunk 1]
```diff
@@ -35,6 +35,20 @@
...
+  {% if page.author.name %}
+    {% assign seo_author_name = page.author.name %}
+  {% else %}
+    {% assign seo_author_name = page.author %}
+  {% endif %}
+
+  {% if page.author.twitter %}
+    {% assign seo_author_twitter = page.author.twitter | replace:"@","" | prepend:"@" %}
+  {% else %}
+    {% assign seo_author_twitter = page.author | replace:"@","" | prepend:"@" %}
```
Comment (benbalter) [inline lib/template.html]: 1. It looks like you're adding the `@`? I believe the Twitter spec says it shouldn't have an `@`, but I haven't looked at it in a while. 

2. We can DRY things up a bit by moving the normalization down to the one place it's used.

Comment (pathawks) [inline lib/template.html]: > It looks like you're adding the `@`?

[It was there before.](https://github.com/benbalter/jekyll-seo-tag/pull/26/files#diff-b18f6b02baa17fff6667f45aca6143c5L101)

The [example at the bottom of this page](https://dev.twitter.com/cards/getting-started) includes the `@`

**Comment by pathawks** [comment]
> We can DRY things up a bit by moving the normalization down to the one place it's used.

All of our other variables are normalized before they are used, and I wanted to keep with that convention.

I could do something like this:

```html
{% if page.author.twitter %}
  {% assign seo_author_twitter %}
{% else %}
  {% assign seo_author_twitter %}
{% endif %}
{% if seo_author_twitter %}
  {% assign seo_author_twitter = page.author.twitter | replace:"@","" | prepend:"@" %}
{% endif %}
```

**Comment by benbalter** [comment]
Nice. Sorry for the delay. Also opened https://github.com/benbalter/jekyll-seo-tag/issues/31 to piggy back on this, so that author's metadata can be centralized, rather than in each post.

**Pull request merged**